parser.js

```diff
@@ -229,14 +229,14 @@
       return createValue(kind, codePoint, pos - (value.length + fromOffset), pos);
     }
 
-    function createCharacter(matches) {
+    function createCharacter(matches, insideClass) {
       var _char = matches[0];
       var first = _char.charCodeAt(0);
       if (hasUnicodeFlag) {
-        if (_char === '}') {
+        if (!insideClass && _char === '}') {
           bail("unescaped or unmatched closing brace");
         }
-        if (_char === ']') {
+        if (!insideClass && _char === ']') {
```

Comment:
Doesn't this change prevent this regex from throwing? `/[0-]]/u`

Reply:
Here is the result when running it:

```
$ regjsparser.parse('[0-]]', 'u')
Uncaught SyntaxError: unescaped or unmatched closing bracket at position 5 (parser.js:1129)
    [0-]]
         ^
    at bail (parser.js:1129)
    at createCharacter (parser.js:240)
    at parseAtom (parser.js:630)
    at parseTerm (parser.js:490)
    at parseAlternative (parser.js:463)
    at parseDisjunction (parser.js:443)
    at Object.parse (parser.js:1144)
    at <anonymous>:1:13
```

It basically parses the class range correctly `0-]` and then complains about the single `]` character.